### PR TITLE
Some changes regarding CssHeaderItem / push integration

### PR DESCRIPTION
**Pull request by klopfdreh**
* there are some methods to apply the rel attribute to css header items
* fixed a bug that the resources were not pushed correctly to the client

### CODE REVIEW

File: wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java [hunk 1]
```diff
@@ -361,7 +366,32 @@ else if (url.toString().startsWith("."))
...
+				String filterPath = WebApplication.get().getWicketFilter().getFilterPath();
+				if (filterPath.equals("/"))
+				{
+					filterPath = "";
+				}
+				else if (filterPath.endsWith("/"))
+				{
+					filterPath = filterPath.replaceAll(".$", "");
+				}
+				partialUrl.append(filterPath);
```
Comment (klopfdreh) [inline wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java]: @martin-g - long time ago we wrote to each other 😄 - I have a question regarding receiving the context path / filter path - is there a method to do so?

Example:

https://127.0.0.1:8443/mywebappcontext/myfilter/wicket/....

I need a better way to get this part:

/mywebappcontext/myfilter/

Comment (martin-g) [inline wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java]: I miss you too! ;-)

`org.apache.wicket.protocol.http.servlet.ServletWebRequest` has methods for both.

**Comment by klopfdreh** [comment]
All in all there are currently some errors regarding the caching of the resources which are pushed. It currently behaves like this:

Resource pushed (no status code) > 200 OK (served via link tag) > 304 NOT MODIFIED (received from cache)

It should be:

Resource pushed (no status code) > 304 NOT MODIFIED (received from cache)

File: wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java (hunk 1 above)
Comment (klopfdreh) [inline wicket-experimental/wicket-http2/wicket-http2-core/src/main/java/org/apache/wicket/http2/markup/head/PushHeaderItem.java]: I used those:
```java
WebApplication.get().getServletContext().getContextPath();
WebApplication.get().getWicketFilter().getFilterPath();
```

and build the path by my self.

Can you have a look at the implementation if this is ok?

**Comment by martin-g** [comment]
This PR should be split into two parts:
- one for the `rel` attribute
- another for the Push related change

Please remove the re-formatting of the code.

**Comment by klopfdreh** [comment]
Closing this PR and replace it with two separated PRs:

https://github.com/apache/wicket/pull/227
https://github.com/apache/wicket/pull/226

**Pull request closed without merging**